### PR TITLE
AP_DDS: Add AP_DDS_config.h for feature defines

### PR DESCRIPTION
**Pull request by Ryanf55**
This will be expanded upon for visual odom and other optional features. It's need by https://github.com/ArduPilot/ardupilot/pull/24518

### CODE REVIEW

File: libraries/AP_DDS/AP_DDS_config.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL_Boards.h>
+
+// UDP only on SITL for now
+#ifndef AP_DDS_UDP_ENABLED
+#define AP_DDS_UDP_ENABLED (CONFIG_HAL_BOARD == HAL_BOARD_SITL)
```
Comment (Ryanf55) [inline libraries/AP_DDS/AP_DDS_config.h:8]: To test: Make sure you can use UDP.

**Review by tridge — APPROVED** [review]

File: libraries/AP_DDS/AP_DDS_config.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <AP_Common/AP_Common.h>
```
Comment (peterbarker) [inline libraries/AP_DDS/AP_DDS_config.h:3]:
```suggestion
```

Bad idea in these _config.h files

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_DDS/AP_DDS_config.h (hunk 2 above)
Comment (Ryanf55) [reply libraries/AP_DDS/AP_DDS_config.h]: Fixed

File: libraries/AP_DDS/AP_DDS_config.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#pragma once
+
```
Comment (peterbarker) [inline libraries/AP_DDS/AP_DDS_config.h:2]:
```suggestion
#include <AP_HAL/AP_HAL_Boards.h>

```

**Review by peterbarker — DISMISSED** [review]

File: libraries/AP_DDS/AP_DDS_config.h (hunk 3 above)
Comment (Ryanf55) [reply libraries/AP_DDS/AP_DDS_config.h:2]: Added.

**Pull request merged**